x/programs/runtime/call_context.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package runtime
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/codec"
+)
+
+var errCannotOverwrite = errors.New("trying to overwrite set field")
+
+type CallContext struct {
+	r               *WasmRuntime
+	defaultCallInfo CallInfo
+}
+
+func (c CallContext) createCallInfo(callInfo *CallInfo) (*CallInfo, error) {
+	newCallInfo := *callInfo
+	resultInfo := reflect.ValueOf(&newCallInfo)
+	callInfoTypeInfo := reflect.TypeOf(CallInfo{})
+	defaults := reflect.ValueOf(c.defaultCallInfo)
+	for i := 0; i < defaults.NumField(); i++ {
+		defaultField := defaults.Field(i)
+		if !defaultField.IsZero() {
+			resultField := resultInfo.Elem().Field(i)
+			if !resultField.IsZero() {
+				return nil, fmt.Errorf("%w %s", errCannotOverwrite, callInfoTypeInfo.Field(i).Name)
+			}
+			resultField.Set(defaultField)
+		}
```

Comment:
Should we not set a "default" value if the default has not been set ?
For instance, that would allow us to set a `CallContext` with a default `FunctionName` and `Fuel` with an `Actor` that *may* be overridden.
If it's not set, I believe that it would be undefined behaviour because then the non-default fields are going to be `nil`.

Reply:
I guess I would need to see how it would integrate into the `programExecuteFunc` function